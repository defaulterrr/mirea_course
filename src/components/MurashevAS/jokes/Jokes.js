import React from 'react';
import QuoteBox from './QuoteBox';

const title = {
  color: 'white',
  textAlign: 'center',
  marginTop: '3rem',
  fontSize: '14px',
};

const joke = 'A void element tag walks into the bar, but the barmen says: "You are not allowed to have children!"';

function Jokes() {
  return (
    <div className="App" style={{ width: '100%', height: '100%' }}>
      <h1 style={title}>{joke}</h1>
      <QuoteBox />
    </div>
  );
}

export default Jokes;
