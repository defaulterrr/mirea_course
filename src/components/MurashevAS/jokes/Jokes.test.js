import React from 'react';
import renderer from 'react-test-renderer';
import Jokes from './Jokes';

describe('Тестирование рендринга', () => {
  test('renders correctly', () => {
    const renderedApp = renderer.create(<Jokes />);
    expect(renderedApp.toJSON()).toMatchSnapshot();
  });
});
