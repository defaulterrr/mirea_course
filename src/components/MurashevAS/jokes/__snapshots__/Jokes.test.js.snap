// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Тестирование рендринга renders correctly 1`] = `
<div
  className="App"
  style={
    Object {
      "height": "100%",
      "width": "100%",
    }
  }
>
  <h1
    style={
      Object {
        "color": "white",
        "fontSize": "14px",
        "marginTop": "3rem",
        "textAlign": "center",
      }
    }
  >
    A void element tag walks into the bar, but the barmen says: "You are not allowed to have children!"
  </h1>
  <div
    className="quote-box"
    id="quote-box"
    style={
      Object {
        "width": "100%",
      }
    }
  >
    <div
      className="quote-container"
      style={
        Object {
          "alignItems": "center",
          "display": "flex",
          "flexFlow": "row wrap",
          "height": 280,
          "justifyContent": "center",
        }
      }
    >
      <h2
        className="setup"
        id="setup"
        style={
          Object {
            "display": "none",
          }
        }
      >
        
      </h2>
      <h2
        className="punchline"
        id="punchline"
        style={
          Object {
            "background": "white",
            "borderRadius": "50%",
            "fontSize": "20px",
            "margin": "1.5rem",
            "padding": "3rem 4rem",
            "textAlign": "center",
            "width": 300,
          }
        }
      >
        
      </h2>
    </div>
    <div
      className="button-container"
      style={
        Object {
          "display": "flex",
          "justifyContent": "center",
        }
      }
    >
      <div
        className="button"
        onClick={[Function]}
        style={
          Object {
            "background": "darkblue",
            "border": "3px solid white",
            "borderRadius": "0.5rem",
            "color": "white",
            "cursor": "pointer",
            "fontSize": "1.2rem",
            "margin": "1rem",
            "padding": "0.5rem",
          }
        }
      >
        Get Answer
      </div>
    </div>
  </div>
</div>
`;
